src/bind-generated.ts

```diff
@@ -0,0 +1,76 @@
+import { UnbindFn, Binding } from './types'
+
+export function bind<Target extends DocumentAndElementEventHandlers, Type extends keyof DocumentAndElementEventHandlersEventMap>(target: Target, { type, listener, options }: Binding<Type, DocumentAndElementEventHandlersEventMap[Type]>): UnbindFn;
+export function bind<Target extends ServiceWorkerRegistration, Type extends keyof ServiceWorkerRegistrationEventMap>(target: Target, { type, listener, options }: Binding<Type, ServiceWorkerRegistrationEventMap[Type]>): UnbindFn;
+export function bind<Target extends XMLHttpRequestEventTarget, Type extends keyof XMLHttpRequestEventTargetEventMap>(target: Target, { type, listener, options }: Binding<Type, XMLHttpRequestEventTargetEventMap[Type]>): UnbindFn;
+export function bind<Target extends AudioScheduledSourceNode, Type extends keyof AudioScheduledSourceNodeEventMap>(target: Target, { type, listener, options }: Binding<Type, AudioScheduledSourceNodeEventMap[Type]>): UnbindFn;
+export function bind<Target extends SpeechSynthesisUtterance, Type extends keyof SpeechSynthesisUtteranceEventMap>(target: Target, { type, listener, options }: Binding<Type, SpeechSynthesisUtteranceEventMap[Type]>): UnbindFn;
+export function bind<Target extends ServiceWorkerContainer, Type extends keyof ServiceWorkerContainerEventMap>(target: Target, { type, listener, options }: Binding<Type, ServiceWorkerContainerEventMap[Type]>): UnbindFn;
+export function bind<Target extends MSInputMethodContext, Type extends keyof MSInputMethodContextEventMap>(target: Target, { type, listener, options }: Binding<Type, MSInputMethodContextEventMap[Type]>): UnbindFn;
+export function bind<Target extends RTCSrtpSdesTransport, Type extends keyof RTCSrtpSdesTransportEventMap>(target: Target, { type, listener, options }: Binding<Type, RTCSrtpSdesTransportEventMap[Type]>): UnbindFn;
+export function bind<Target extends GlobalEventHandlers, Type extends keyof GlobalEventHandlersEventMap>(target: Target, { type, listener, options }: Binding<Type, GlobalEventHandlersEventMap[Type]>): UnbindFn;
+export function bind<Target extends WindowEventHandlers, Type extends keyof WindowEventHandlersEventMap>(target: Target, { type, listener, options }: Binding<Type, WindowEventHandlersEventMap[Type]>): UnbindFn;
+export function bind<Target extends HTMLFrameSetElement, Type extends keyof HTMLFrameSetElementEventMap>(target: Target, { type, listener, options }: Binding<Type, HTMLFrameSetElementEventMap[Type]>): UnbindFn;
```

Comment:
Tried another approach - code generation. However, while writing this, I realized that it'd be better to just generate `.d.ts` files instead of all code.

Reply:
What did you use to generate this?

Reply:
*Ideally* if we can avoid generation that is great because it means we don't need to keep bumping as the typescript types change

Reply:
I generate it in `generate.js` file.